src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.cs

```diff
@@ -503,6 +503,41 @@ internal static IEnumerable<string> GetFrameworkAssembliesToTreatAsUserAssemblie
 			return acw_map;
 		}
 
+		public static Dictionary<string, HashSet<string>> LoadMapFile (IBuildEngine4 engine, string mapFile)
+		{
+			var cachedMap = (Dictionary<string, HashSet<string>>)engine?.GetRegisteredTaskObject (mapFile, RegisteredTaskObjectLifetime.Build);
+			if (cachedMap != null)
+				return cachedMap;
+			var map = new Dictionary<string, HashSet<string>> ();
+			if (!File.Exists (mapFile))
+				return map;
+			foreach (var s in File.ReadLines (mapFile)) {
+				var items = s.Split (';');
+				if (!map.ContainsKey (items [0]))
+					map.Add (items [0], new HashSet<string> ());
+				if (!map[items [0]].Contains (items [1]))
+					map [items [0]].Add (items [1]);
+			}
```

Comment:
If there are a lot of items in this dictionary, we should use some local variable so indexers aren't called as much.

Would help perf a little to be something like:
```csharp
foreach (var s in File.ReadLines (mapFile)) {
    var split = s.Split (';');
    var key = split [0];
    var value = split [1];
    if (!map.TryGetValue (key, out HashSet<string> items)) {
        map [key] = items = new HashSet<string> ();
    }
    if (!items.Contains (value)) {
        items.Add (value);
    }
}
```